### PR TITLE
[full-ci] Add switch to collapse/expand sidebar

### PR DESCRIPTION
**Pull request by paulcod3**
## Related Issue

- Fixes https://github.com/owncloud/web/issues/3919
- Fixes https://github.com/owncloud/web/issues/5165
- Fixes https://github.com/owncloud/web/issues/2061 (already closed, but for reference)
- Fixes https://github.com/owncloud/web/issues/5666

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPublicManagement</strong> https://drone.owncloud.com/owncloud/web/18310/34/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by paulcod3** [comment]
sidebarclosed
sidebaropen

**Comment by tbsbdr** [comment]
screenshots: some sidebars with icons i found.
I asked Simone and she said, the info-i would be missleading, she would expect a double-arrow. @paulcod3 agree, that the material double arrow is ugly (https://fonts.google.com/icons?selected=Material%20Icons%3Adouble_arrow%3A). 

there is a nice community icon from https://materialdesignicons.com/icon/chevron-double-right that i would prefer. what do you think @paulcod3 

image
image
image
image
image

File: packages/web-app-files/src/components/AppBar/ViewOptions.vue [hunk 1]
```diff
@@ -10,8 +10,18 @@
       size="small"
       gap-size="xsmall"
     >
-      <oc-icon name="tune" size="small" />
-      <translate>View</translate>
+      <oc-icon name="tune" size="medium" />
+    </oc-button>
+    <oc-button
+      :aria-label="viewButtonAriaLabel"
```
Comment (kulmann) [inline packages/web-app-files/src/components/AppBar/ViewOptions.vue:16]: aria label references the computed prop of the other button

File: packages/web-app-files/src/components/SideBar/SideBar.vue [hunk 2]
```diff
@@ -161,6 +166,10 @@ export default {
         })
       },
       immediate: true
+    },
+    currentFolder(now, old) {
```
Comment (kulmann) [inline packages/web-app-files/src/components/SideBar/SideBar.vue:170]: dev leftover

File: packages/web-app-files/src/components/SideBar/SideBar.vue [hunk 3]
```diff
@@ -1,9 +1,10 @@
 <template>
   <div
-    v-click-outside="onClickOutside"
     :class="{
       'has-active': !!appSidebarActivePanel
     }"
+    @click.stop
+    v-if="highlightedFile != null"
```
Comment (kulmann) [inline packages/web-app-files/src/components/SideBar/SideBar.vue:7]:
```suggestion
    v-if="highlightedFile !== null"
```

File: packages/web-app-files/src/components/SideBar/SideBar.vue [hunk 4]
```diff
@@ -151,6 +153,9 @@ export default {
     },
     areMultipleSelected() {
       return this.selectedFiles && this.selectedFiles.length > 1
+    },
+    isRootFolder() {
+      return this.highlightedFile.path === ""
```
Comment (kulmann) [inline packages/web-app-files/src/components/SideBar/SideBar.vue:158]:
```suggestion
      !return this.highlightedFile?.path
```

**Review by kulmann — CHANGES_REQUESTED** [review]
- Upon selection of multiple files, I get a `Duplicate keys detected: 'panel-details-item'. This may cause an update error.` in the JS console.

See comments for some other minor issues.
UX feels AWESOME! Love it! 😍

**Comment by pascalwengerter** [comment]
Agreed, UX feels neat!

Upon page reload, there's two errors in the console though:
- TypeError: this.highlightedFile is null
- TypeError: this.$refs.panels is undefined

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/18396/28/1
The following scenarios passed on retry:
- webUISharingInternalUsersToRootSharingIndicator/shareWithUsers.feature:98

**Comment by ownclouders** [comment]
Results for <strong>oC10XGAPortrait1</strong> https://drone.owncloud.com/owncloud/web/18396/42/1
The following scenarios passed on retry:
- webUICreateFilesFolders/createFolderEdgeCases.feature:60
- webUICreateFilesFolders/createFolderEdgeCases.feature:61
- webUICreateFilesFolders/createFolderEdgeCases.feature:62

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingExternalRoot</strong> https://drone.owncloud.com/owncloud/web/18396/40/1
The following scenarios passed on retry:
- webUISharingExternalToRoot/federationSharing.feature:356

**Comment by tbsbdr** [comment]
nice one :-)

sidebar should be colsed initially

**Comment by paulcod3** [comment]
> Agreed, UX feels neat!
> 
> Upon page reload, there's two errors in the console though:
> 
> * TypeError: this.highlightedFile is null
> * TypeError: this.$refs.panels is undefined
For me I don't get this error anymore since I check for null, maybe you had an old version?

**Comment by pascalwengerter** [comment]
> > Agreed, UX feels neat!
> > Upon page reload, there's two errors in the console though:
> > 
> > * TypeError: this.highlightedFile is null
> > * TypeError: this.$refs.panels is undefined
> 
> For me I don't get this error anymore since I check for null, maybe you had an old version?

Just pulled on your branch and still get the second one 🤔

**Review by kulmann — CHANGES_REQUESTED** [review]
Needs a changelog item

**Comment by kulmann** [comment]
> screenshots: some sidebars with icons i found.
> I asked Simone and she said, the info-i would be missleading, she would expect a double-arrow. @paulcod3 agree, that the material double arrow is ugly (https://fonts.google.com/icons?selected=Material%20Icons%3Adouble_arrow%3A).
> 
> there is a nice community icon from https://materialdesignicons.com/icon/chevron-double-right that i would prefer. what do you think @paulcod3

We used that community double chevron icons. Thanks for the pointer 👍

**Review by kulmann — CHANGES_REQUESTED** [review]
`click outside` handler doesn't work/exist anymore. closing the sidebar is not possible anymore with a click on whitespace. that one is important.

File: packages/web-app-files/src/store/sidebar.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+export default {
+  state: () => ({
+    sidebarClosed: true
```
[the remark this answers is not in the document]
Comment (kulmann) [reply packages/web-app-files/src/store/sidebar.js:3]: reasoning: this store module is only about the sidebar. no need to mention the word `sidebar` inside the store module in any way.

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/18449/12/1
The following scenarios passed on retry:
- webUIFilesActionMenu/versions.feature:36

**Comment by ownclouders** [comment]
Results for <strong>oC10XGAPortrait1</strong> https://drone.owncloud.com/owncloud/web/18451/42/1
The following scenarios passed on retry:
- webUIDeleteFilesFolders/deleteFilesFolders.feature:59

File: packages/web-app-files/src/components/SideBar/SideBar.vue [hunk 6]
```diff
@@ -1,9 +1,9 @@
 <template>
   <div
-    v-click-outside="onClickOutside"
     :class="{
       'has-active': !!appSidebarActivePanel
     }"
+    @click.stop
```
Comment (kulmann) [inline packages/web-app-files/src/components/SideBar/SideBar.vue:6]: why is this line needed?

File: packages/web-app-files/src/views/PublicFiles.vue [hunk 7]
```diff
@@ -142,6 +135,10 @@ export default {
     }
   },
 
+  data: () => ({
+    loading: true
```
Comment (kulmann) [inline packages/web-app-files/src/views/PublicFiles.vue:139]: `loading` is not referenced, right? Then you can delete it again. dev leftover?

Comment (paulcod3) [reply packages/web-app-files/src/views/PublicFiles.vue:139]: yeah it is, its for the list-loader

**Review by kulmann — APPROVED** [review]
Awesome! 😍

**Comment by kulmann** [comment]
Please squash on merge ;-)
